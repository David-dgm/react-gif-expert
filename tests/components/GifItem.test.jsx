import { render,screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Prueba Componente GifItem', () => {
    const title = 'Saitama';
    const url = 'http://direccion/mock';

    test('should match with spanshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('should show image with url and requested alt', () => {
        render(<GifItem title={title} url={url} />);
        
        const {src , alt}=screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

     test('should show title', () => {
         render(<GifItem title={title} url={url} />);

         expect(screen.getByText( title)).toBeTruthy();

        //  expect(screen.getByTestId('p-title'))
        //  .toMatch({ title });
        //  .toContainEqual({ title });
         
     });
});
