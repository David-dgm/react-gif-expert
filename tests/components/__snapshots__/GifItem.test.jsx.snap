// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Componente GifItem should match with spanshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://direccion/mock"
    />
    <p
      data-testid="p-title"
    >
      Saitama
    </p>
  </div>
</div>
`;
